### PR TITLE
scripts: west_commands: handle EOFError in nrfjprog.py

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
We should simply exit if the user hits control-D during the prompt.

### CODE REVIEW

**Review by oyvindronningstad — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Comment by ioannisg** [comment]
@tejlmand take a quick look?

**Review by ioannisg — APPROVED** [review]

File: scripts/west_commands/runners/nrfjprog.py [hunk 1]
```diff
@@ -111,7 +111,10 @@ def get_board_snr(self, glob):
         p = 'Please select one with desired serial number (1-{}): '.format(
                 len(snrs))
         while True:
-            value = input(p)
+            try:
+                value = input(p)
+            except EOFError:
+                sys.exit(0)
```
Comment (tejlmand) [inline scripts/west_commands/runners/nrfjprog.py:117]: This only works if `CTRL-D` is pressed when waiting for the first character.

In this case:
```
Please select one with desired serial number (1-2): 1<CTRL>-D
```
then nothing happens.

this works:
```
Please select one with desired serial number (1-2): <CTRL>-D
```

**Review by tejlmand — APPROVED** [review]
Working, as long as `<CTRL>-D` is pressed while expecting first character.

Approving as this PR is still an improvement.

File: scripts/west_commands/runners/nrfjprog.py (hunk 1 above)
Comment (ioannisg) [reply scripts/west_commands/runners/nrfjprog.py:117]: @mbolivar-nordic  you might want to address this too.

Comment (mbolivar-nordic) [reply scripts/west_commands/runners/nrfjprog.py:117]: The input() function is part of the python standard library. It doesn't return control to the user on the first control-d if characters have already been typed.

You have to keep hitting control-D multiple times to get it to exit. Then you can hit control-D again when it reprompts, like this:

```
$ west flash
[...]
There are multiple boards connected.
[...]
Please select one with desired serial number (1-2): xxx^D^D^DPlease select one with desired serial number (1-2): ^D
$
```

If you've already entered some characters, control-C already works. I'm just adding control-D support in this PR.

**Pull request merged**